File: 908-middle-of-the-linked-list/middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        int len=0;
        ListNode mid=head;
        while(head!=null){
            len++;
            head=head.next;
        }
        for(int i=0;i<len/2;i++) mid=mid.next;
        return mid;
    }
}

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* mid=head;
        int len=0;
        while(head!=nullptr){
            len++;
            head=head->next;
        }
        for(int i=0;i<len/2;i++) mid=mid->next;
        return mid;
        
    }
};